assets/js/app.js

```diff
@@ -0,0 +1,277 @@
+                button.addEventListener('click', async function () {
+
+                    //save the data to cache key + storage
+                    const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+                    const loadingDiv = document.getElementById("loadingDiv")
+                    const index = coinsCache.findIndex(coin => coin.id === this.id)
+                    let coinData = ``
+                    if (index !== -1) coinData = coinsCache[index]
+                    else {
+                        loadingDiv.classList.add(`visible`)
+                        coinData = await getSingleCoin(this.id)
+                        loadingDiv.classList.remove(`visible`)
+                    }
+
+                    if (!coinsCache.some(coin => coin.id === coinData.id)) coinsCache.push(coinData)
+                    localStorage.setItem(`coinsCache`, JSON.stringify(coinsCache))
+
+                    //after 2 min, deletes from the cache (so in the next call, get the most updated data)
+                    setTimeout(() => {
+                        const coinsCache = JSON.parse(localStorage.getItem(`coinsCache`))
+                        const index = coinsCache.findIndex(coin => coin.id = this.id)
+                        coinsCache.splice(index, 1)
+                        localStorage.setItem(`coinsCache`, JSON.stringify(coinsCache))
+                    }, 120000)
+
+                    //creates the inner html of the popover
+                    button.setAttribute('data-bs-content', `
+                    <img src="${coinData.image.thumb}" />
+                    <p>${coinData.name}</p>
+                    <p>usd: ${coinData.market_data.current_price.usd}$</p>
+                    <p>eur: ${coinData.market_data.current_price.eur}€</p>
+                    <p>ils: ${coinData.market_data.current_price.ils}₪</p>
+                `)
+                    // show the popover if there isn`t and hide if there is
+                    if (!button._popover) {
+                        button._popover = new bootstrap.Popover(button, {
+                            trigger: 'manual',
+                            html: true
+                        })
+                        // Show the updated popover
+                        button._popover.show()
+                    }
+                    else button._popover.hide()
+                })
+            })
+
+            //the saving of the 5 coins
+            const save5coins = document.querySelectorAll("div > input")
+            save5coins.forEach(function (input) {
+                input.addEventListener(`change`, function () { //invoke on every change of checkbox
+                    const fiveCoinsInStorage = JSON.parse(localStorage.getItem(`5coinsInStorage`))
+                    if (input.checked) { //if the change was to ON
+                        if (fiveCoinsInStorage.length === 5) { //if we already have 5 coins saved
```

Comment:
what happens if the user closes the modal without deselecting a coin:
<img width="805" alt="Screen Shot 2025-01-29 at 13 01 22" src="https://github.com/user-attachments/assets/623e857d-08e9-4cd1-bd43-9933912ab2f3" />